### PR TITLE
EE-6510: Jenkins Migration Phase 1 - 3.x: Migrate AMQP Release Jobs

### PR DESCRIPTION
**Pull request by garodriguezlp**
- Changes for publishing to Exchange

### CODE REVIEW

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+String nexusRepositoryId = "mule-ee-releases"
+String nexusRepositoryUrl = "https://repository-master.mulesoft.org/nexus-ee/content/repositories/releases/"
+
+String[] parts = environments["${deployment_environment_param}"].split("::")
+String exchangeMvnRepositoryId = parts[0]
+String exchangeMvnRepositoryUrl = parts[1]
+
+node('ubuntu-14.04') {
+    try {
+//    notifySlack('STARTED', slack_channel)
```
Comment (aiannucci) [inline release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Remove comments

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+//            mvn "org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy-file -X" +
+//                    " -DgroupId=$groupIdStudioPlugin" +
+//                    " -DartifactId=$artifactIdStudioPlugin" +
+//                    " -Dversion=$version" +
+//                    " -Dpackaging=zip" +
+//                    " -Dfile=$artifactIdStudioPlugin-$version-studio-plugin.zip" +
+//                    " -Dclassifier=studio-plugin" +
+//                    " -DpomFile=$artifactIdStudioPlugin-$version.pom" +
+//                    " -DrepositoryId=$exchangeMvnRepositoryId" +
+//                    " -Durl=$exchangeMvnRepositoryUrl"
```
Comment (aiannucci) [inline release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Remove comments

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+//            mvn "org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy-file -X" +
+//                    " -DgroupId=$groupIdJar" +
+//                    " -DartifactId=$artifactId" +
+//                    " -Dversion=$version" +
+//                    " -Dpackaging=jar" +
+//                    " -Dfile=$artifactId-$version.jar" +
+//                    " -DgeneratePom=false" +
+//                    " -DrepositoryId=$exchangeMvnRepositoryId" +
+//                    " -Durl=$exchangeMvnRepositoryUrl"
```
Comment (aiannucci) [inline release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Remove comments

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        throw e
+    } finally {
+
+        stage('Clean workspace') {
+            deleteDir()
+        }
+
+//    notifySlack(currentBuild.result, slack_channel)
```
Comment (aiannucci) [inline release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Remove comments

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    }
+}
```
Comment (aiannucci) [inline release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Do we really need to keep all this duplicated logic here? Why can't we reuse it from a shared library?
Also, remember this project is public. Doesn't sound like something we should keep here.

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+String version = new_version_param
+
+String nexusRepositoryId = "mule-ee-releases"
+String nexusRepositoryUrl = "https://repository-master.mulesoft.org/nexus-ee/content/repositories/releases/"
+
+String[] parts = environments["${deployment_environment_param}"].split("::")
+String exchangeMvnRepositoryId = parts[0]
+String exchangeMvnRepositoryUrl = parts[1]
+
+node('ubuntu-14.04') {
```
Comment (aiannucci) [inline release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Upgrade to 18.04

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy (hunk 5 above)
Comment (garodriguezlp) [reply release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: Good point. My criteria for leaving this here was to be close to the rest of the release pipeline files. But now that you mentioned, I agree, too much sensitive information here. 

Where would be the best place to leave it? Should we move it alone or with the rest of the scripts for the release pipeline?

File: release-pipeline/Jenkinsfile-upload-to-exchange.groovy (hunk 1 above)
Comment (garodriguezlp) [reply release-pipeline/Jenkinsfile-upload-to-exchange.groovy]: All the commented code here left intentionally for test purposes. The main idea with the PR was to validate my whole approach, to then move on the details. 

Referring to that ☝️... does it look fine regarding its intended purpose?

**Pull request closed without merging**